File: EasyMusic/lyricswidget.cpp
#include "lyricswidget.h"
#include <QPainter>

LyricsWidget::LyricsWidget(QWidget *parent)
    : QWidget{parent}
{

}

void LyricsWidget::setLyrics(const QString &pre_, const QString &cur_, const QString &next_)
{
    previous = pre_;
    current = cur_;
    next = next_;
    update();
}

void LyricsWidget::paintEvent(QPaintEvent *event)
{
    QPainter painter(this);

    // painter.setBrush(QColor("#FF66CC"));

    painter.fillRect(rect(),QColor(0,0,0,0));
    int centerY = height()/2;
    int lineHeight = 70;

    QFont preFont;
    preFont.setPointSize(18);
    painter.setFont(preFont);
    painter.setPen(QColor("#FF66CC"));
    painter.drawText(QRect(0, centerY - lineHeight, width(), lineHeight),Qt::AlignCenter,previous);


    QFont currentFont;
    currentFont.setPointSize(25);
    currentFont.setBold(true);
    painter.setFont(currentFont);
    painter.setPen(QColor("#FF0033"));
    painter.drawText(QRect(0, centerY, width(), lineHeight), Qt::AlignCenter, current);

    QFont nextFont;
    nextFont.setPointSize(18);
    painter.setFont(nextFont);
    painter.setPen(QColor("#FF3399"));
    painter.drawText(QRect(0, centerY + lineHeight, width(), lineHeight), Qt::AlignCenter, next);

}

File: EasyMusicQt/lyricswidget.py
from PyQt6.QtWidgets import QWidget
from PyQt6.QtGui import QPainter, QColor, QFont
from PyQt6.QtCore import Qt


class LyricsWidget(QWidget):
    def __init__(self, parent=None):
        super().__init__(parent)
        self.previous = ""
        self.current = ""
        self.next = ""

    def set_lyrics(self, pre_text, cur_text, next_text):
        self.previous = pre_text
        self.current = cur_text
        self.next = next_text
        self.update()

    def paintEvent(self, event):
        painter = QPainter(self)

        # 透明背景
        painter.fillRect(self.rect(), QColor(0, 0, 0, 0))

        center_y = self.height() // 2
        line_height = 70

        # 上一行歌词
        pre_font = QFont()
        pre_font.setPointSize(18)
        painter.setFont(pre_font)
        painter.setPen(QColor("#FF66CC"))
        painter.drawText(0, center_y - line_height, self.width(), line_height,
                         Qt.AlignmentFlag.AlignCenter, self.previous)

        # 当前行歌词
        current_font = QFont()
        current_font.setPointSize(25)
        current_font.setBold(True)
        painter.setFont(current_font)
        painter.setPen(QColor("#FF0033"))
        painter.drawText(0, center_y, self.width(), line_height,
                         Qt.AlignmentFlag.AlignCenter, self.current)

        # 下一行歌词
        next_font = QFont()
        next_font.setPointSize(18)
        painter.setFont(next_font)
        painter.setPen(QColor("#FF3399"))
        painter.drawText(0, center_y + line_height, self.width(), line_height,
                         Qt.AlignmentFlag.AlignCenter, self.next)